contrib/pylightning/lightning/plugin.py

```diff
@@ -373,16 +370,19 @@ def _dispatch_notification(self, request):
         except Exception:
             self.log(traceback.format_exc())
 
+    def _write_locked(self, obj):
```

Comment:
Yeah, I definitely checked before and after.  If that JSON seralization fails, you get nothing; after this patch it gave the expected error.

I was surprised too, because surely an exception should drop the lock?

Reply:
Ok, this intrigues me. Tried to reproduce but couldn't:

```python
>>> import json
>>> from threading import Lock
>>> l = Lock()
>>> l.locked()
False
>>> with l:
...     json.loads('{{{')
...
Traceback (most recent call last):
  File "<stdin>", line 2, in <module>
  File "/usr/lib/python3.6/json/__init__.py", line 354, in loads
    return _default_decoder.decode(s)
  File "/usr/lib/python3.6/json/decoder.py", line 339, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
  File "/usr/lib/python3.6/json/decoder.py", line 355, in raw_decode
    obj, end = self.scan_once(s, idx)
json.decoder.JSONDecodeError: Expecting property name enclosed in double quotes: line 1 column 2 (char 1)
>>> l.locked()
False
>>> l.acquire()                                                                                                                                                                                                                                                                           
True                                                                                                                                                                                                                                                                                      
>>>
```

Really strange, I can't explain it to myself.

Reply:
It could have been that `json.loads()` triggers a special exception that bypasses context releases (sort of unchecked exceptions in Java), but that doesn't seem to be the case.

Happy to apply this, but not being able to reproduce makes me uneasy :wink: